docs/advanced-usage/nesting-aggregate-metrics.md

```diff
@@ -0,0 +1,119 @@
+---
+icon: kolena/property-16
+---
+
+# :kolena-property-20: Nesting Aggregate Metrics
+
+When computing [test case metrics][kolena.workflow.MetricsTestCase] in an
+[evaluator](../reference/workflow/evaluator.md), in some cases it is desirable to compute multiple sets of aggregate
+metrics within a given test case. Here are a few examples of scenarios where this pattern might be warranted:
+
+- **Multiclass workflows**: for ML tasks with multiple classes, a given test case may contain samples from
+    more than one class. While it's useful to report metrics aggregated across all classes using an
+    [averaging method](../metrics/averaging-methods.md), it's also useful to see aggregate metrics computed for each of
+    the classes.
+- **Ensemble or pipeline models**: when testing an ensemble or a pipeline containing multiple models, it can be useful
+    to see metrics from the output of the complete ensemble/pipeline  as well as metrics computed for each of the
+    constituent models.
+
+In these cases, it is possible to nest additional aggregate metrics records within a
+[`MetricsTestCase`][kolena.workflow.MetricsTestCase] object returned from an evaluator. In this tutorial, we'll learn
+how to use this API to report class-level or other nested aggregate metrics within a test case.
+
+## Example: Multiclass Object Detection
+
+Let's consider the case of a multiclass object detection task with objects of type `Airplane`, `Boat`, and `Car`.
+When a test case contains images with each of these three classes, test-case-level metrics are the average (e.g.
+[micro](../metrics/averaging-methods.md#micro-average), [macro](../metrics/averaging-methods.md#macro-average), or
+[weighted](../metrics/averaging-methods.md#weighted-average)) of class-level metrics across each of these three classes.
+
+Using [macro-averaged](../metrics/averaging-methods.md#macro-average) precision, recall, and F1 score, and mean average
+precision score (mAP) across all images in the test case.
+
+| Test Case | # Images | `Precision_macro` | `Recall_macro` | `F1_macro` | `mAP` |
```

Comment:
I usually see the averaging method first, before the metric eg macro_Precision because it reads "macro-averaged precision"